components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,13 +164,65 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
 	}
 
-	if err := oAx.home(ctx, np); err != nil {
+	if err := oAx.home(context.Background(), np); err != nil {
 		return nil, err
 	}
 
+	oAx.checkHit(context.Background())
 	return oAx, nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	ctx, g.cancelFunc = context.WithCancel(ctx)
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return nil
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			func() {
+				g.mu.Lock()
+				defer g.mu.Unlock()
+				if hit0, _ := g.limitHit(ctx, true); hit0 {
+					g.moveAway(ctx, true)
+				}
+				if hit1, _ := g.limitHit(ctx, false); hit1 {
+					g.moveAway(ctx, false)
+				}
+			}()
+		}
+	})
+}
+
+func (g *oneAxis) moveAway(ctx context.Context, zero bool) {
+	g.motor.Stop(ctx, nil)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Question for reviewers: if I remove this sleep command, I get a context canceled error and the server hangs. It will only stop after a minute and 30 seconds timeout. I want to add a comment explaining why this sleep is necessary but I don't really understand why that would be happening. Any ideas?

Reply:
no idea but I will try and figure out a way to check. good suggestion

Reply:
let's print out if the contexts are actually being called at any time, I am struggling with the same thing on my base code actually.

Reply:
This line needs an explanatory comment one way or the other. Ideally it would be "we need this sleep for the following reason," but in the worst-case scenario it should be "we need this sleep for unknown reasons, because if you take it out the server hangs with errors about a canceled context." 

(I suppose _ideally_ ideally we could remove the sleep entirely, though that sounds like it might be a whole separate task.)